src/helics/core/CommonCore.cpp

```diff
@@ -2277,16 +2280,25 @@ std::string CommonCore::coreQuery(const std::string& queryStr) const
     if (mi != mapIndex.end()) {
         auto index = mi->second.first;
         if (isValidIndex(index, mapBuilders) && !mi->second.second) {
-            if (std::get<0>(mapBuilders[index]).isCompleted()) {
-                return std::get<0>(mapBuilders[index]).generate();
+            auto& builder = std::get<0>(mapBuilders[index]);
+            if (builder.isCompleted()) {
+                auto center = generateMapObjectCounter();
+                if (center == builder.getCounterCode()) {
+                    return std::get<0>(mapBuilders[index]).generate();
```

Comment:
Would this work or is there something happening that makes calling generate() on the builder object not work?
```suggestion
                    return builder.generate();
```